3party/pugixml/utils.hpp

```diff
@@ -1,6 +1,10 @@
 #pragma once
 
+#ifdef SYSTEM_PUGIXML
+#include "pugixml.hpp"
+#else
 #include "pugixml/src/pugixml.hpp"
+#endif
```

Comment:
This can be done in more "natural" way:
Add to 3party/pugixml/CMakeList.txt
```
target_include_directories(pugixml PUBLIC pugixml/src)

add_library(pugixml::pugixml ALIAS pugixml)
```

Then each dependent library can declare in CMakeList.txt just:
```
target_link_libraries(${PROJECT_NAME} pugixml::pugixml)
```
and in source files:
```
#include <pugixml.hpp>
```